tests/test_model_functions.py

```diff
@@ -1,69 +1,510 @@
+            'Derivative(_Ca_independent_transient_outward_K_current_r_gate$r, _environment$time), '\
+            'Derivative(_Ca_independent_transient_outward_K_current_s1_gate$s1, _environment$time), '\
+            'Derivative(_Ca_independent_transient_outward_K_current_s2_gate$s2, _environment$time), '\
+            'Derivative(_Ca_independent_transient_outward_K_current_s3_gate$s3, _environment$time), '\
+            'Derivative(_delayed_rectifier_K_current_z_gate$z, _environment$time), '\
+            'Derivative(_delayed_rectifier_K_current_pa_gate$p_a, _environment$time), '\
+            'Derivative(_delayed_rectifier_K_current_pi_gate$p_i, _environment$time), '\
+            'Derivative(_intracellular_ion_concentrations$Na_i, _environment$time), '\
+            'Derivative(_intracellular_ion_concentrations$Ca_i, _environment$time), '\
+            'Derivative(_intracellular_ion_concentrations$K_i, _environment$time), '\
+            'Derivative(_intracellular_Ca_buffering$O_C, _environment$time), '\
+            'Derivative(_intracellular_Ca_buffering$O_TC, _environment$time), '\
+            'Derivative(_intracellular_Ca_buffering$O_TMgC, _environment$time), '\
+            'Derivative(_intracellular_Ca_buffering$O_TMgMg, _environment$time), '\
+            'Derivative(_cleft_space_ion_concentrations$K_c, _environment$time), '\
+            'Derivative(_Ca_handling_by_the_SR$Ca_rel, _environment$time), '\
+            'Derivative(_Ca_handling_by_the_SR$Ca_up, _environment$time), '\
+            'Derivative(_Ca_handling_by_the_SR$O_Calse, _environment$time), '\
+            'Derivative(_Ca_handling_by_the_SR$F1, _environment$time), '\
+            'Derivative(_Ca_handling_by_the_SR$F2, _environment$time), '\
+            'Derivative(_Ca_handling_by_the_SR$F3, _environment$time)]'
+
+    # also tested by model_units
+    def test_get_equations_for(self, basic_model):
+        """ Tests Model.get_equations_for() works correctly.
+        Note: the basic model has 1 equation dsv11/dt = 1 which is not
+        related to a symbol and so has no equations that can be retrieved
+        by symbol
+        """
+
+        symbol_a = basic_model.get_symbol_by_cmeta_id("sv11")
+        equation = basic_model.get_equations_for([symbol_a])
+        assert len(equation) == 0
+
+        symbol_t = basic_model.get_symbol_by_name("environment$time")
+
+        equation1 = basic_model.get_equations_for([symbol_t])
+        assert len(equation1) == 0
+
+    # also tested by model_units
+    def test_get_equations_for_1(self, aslanidi_model, OXMETA):
+        """ Tests Model.get_equations_for() works correctly. """
+
+        symbol_a = aslanidi_model.get_symbol_by_ontology_term(OXMETA, "membrane_capacitance")
+        equation = aslanidi_model.get_equations_for([symbol_a])
+        assert len(equation) == 1
+        assert equation[0].lhs == symbol_a
+        assert equation[0].rhs == 5e-5
+
+    def test_get_equations_for_2(self, hh_model, OXMETA):
```

Comment:
Out of curiosity: Any particular reason for having this as a fixture rather than just writing `shared.OXMETA` ?

Reply:
Overzealous use of fixtures - changed